src/main/java/frc/robot/commands/ConveyanceIndexCommand.java

```diff
@@ -0,0 +1,44 @@
+package frc.robot.commands;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.Robot;
+
+
+public class ConveyanceIndexCommand extends CommandBase {
+    private boolean isBallBetweenSensors; // ball state
+    
+    public ConveyanceIndexCommand() {
+        addRequirements(Robot.CONVEYANCE_SUBSYSTEM);
+    }
+
+        // Called when the command is initially scheduled.
+    @Override
+    public void initialize() {}
+
+    // Called every time the scheduler runs while the command is scheduled.
+    @Override
+    public void execute() {
+      if (Robot.CONVEYANCE_SUBSYSTEM.getConveyanceHasBall()  && Robot.FEEDER_SUBSYSTEM.getFeederHasBall() ) {
+      Robot.CONVEYANCE_SUBSYSTEM.stopConveyanceOne();   //when there is a ball in conveyance stage 1 and 2 conveyance wont run        
+       }        
+    
+      if (Robot.CONVEYANCE_SUBSYSTEM.getConveyanceHasBall()  && Robot.FEEDER_SUBSYSTEM.getFeederHasBall() == false )    {        
+         Robot.CONVEYANCE_SUBSYSTEM.setConveyanceMaxForward();  //if there is a ball in comveyance stage 1 but nothing at stage 2 conveyance will run at 1
+       }
+   
+      if (Robot.CONVEYANCE_SUBSYSTEM.getConveyanceHasBall() == false ) {
```

Comment:
Use "else if" here, its easier to reason that the first matching condition is what is acted on. In this case though, we act on every matching condition.